pkg/proc/gdbserial/undo.go

```diff
@@ -158,3 +160,107 @@ func UndoIsRecording(recordingFile string) (result bool, err error) {
 
 	return bytes.Equal(marker, data), nil
 }
+
+// Fetch the output of a udbserver get_info command, split on ; and , characters.
+//
+// This is not (currently) implementing a proper parse of the data returned, just making it more
+// convenient to search.
+func undoGetInfo(conn gdbConn) []string {
+	info, err := conn.undoCmd("get_info")
+	if err != nil {
+		panic("Failed to retrieve Undo stop info.")
+	}
+	splitter := func(c rune) bool {
+		return c == ';' || c == ','
+	}
+	return strings.FieldsFunc(info, splitter)
+}
+
+func undoGetLogExtent(p *gdbProcess) (uint64, uint64) {
+	extent, err := p.conn.undoCmd("get_log_extent")
+	if err != nil {
+		panic("Failed to retrieve Undo log extent.")
```

Comment:
Panicking here and elsewhere in the file is storing up trouble for later: udbserver is a separate process, so it could be SIGKILL'd under our feet (for example, by the OOM killer). It would be nice if that kind of situation didn't also crash Delve. What happens if you propagate the error instead of panicking, as you do in `undoWhen`?

We have a similar issue in UDB where a crash in Nanny is not very well handled, but the UDB situation is much worse because of a long legacy of not handling communication errors.

Reply:
My inclination on this was to keep error handling pretty minimal when bad stuff happens (we've also had problems in UDB / Nanny caused by trying to struggle on when weird stuff has already happened).  But I think I was probably  also being lazy in this instance - it doesn't look that hard to just plumb it through, so I'll have a go.

Reply:
This resulted in some more significant restructuring of the patch than I'd have hoped but the end result is, I think, fairly tidy.